java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -1464,8 +1465,74 @@ private int diffVarDef(JCVariableDecl oldT, JCVariableDecl newT, int[] bounds) {
                 addDimensions = dimension(newT.vartype, -1);
                 cLikeArray = vartypeBounds[1] > oldT.pos;
                 cLikeArrayChange =  cLikeArray && dimension(oldT.vartype, oldT.pos) > addDimensions;
-                copyTo(localPointer, vartypeBounds[0]);
+
+                /**
+                 * Extracting modifier from oldTree using symbol position and
+                 * modifier positions when oldT.type is error and vartype
+                 * upperbound is not proper.
+                 */
+                if (oldT.type.getKind() == TypeKind.ERROR && vartypeBounds[1] == -1) {
+
+                    // returns -1 if modifiers not present.
+                    int modsUpperBound = getCommentCorrectedEndPos(oldT.mods);
+                    if (modsUpperBound > -1) {
+                        tokenSequence.move(modsUpperBound);
+                        tokenSequence.moveNext();
+
+                        // copying modifiers from oldTree
+                        if (JavaTokenId.WHITESPACE == tokenSequence.token().id()) {
+                            int offset = tokenSequence.offset();
+                            copyTo(localPointer, localPointer = offset);
+                        } else {
+                            copyTo(localPointer, localPointer = modsUpperBound);
+                        }
+                    }
+
+                    tokenSequence.move(localPointer);
+                    tokenSequence.moveNext();
+                    int offset = tokenSequence.offset();
+                    JavaTokenId tokenId = tokenSequence.token().id();
+
+                    //adding back all whitespaces/block-comment/javadoc-comments present in OldTree before variable type token.
+                    while ((tokenId == JavaTokenId.WHITESPACE || tokenId == JavaTokenId.BLOCK_COMMENT || tokenId == JavaTokenId.JAVADOC_COMMENT) && offset < oldT.sym.pos) {
+                        tokenSequence.moveNext();
```

Comment:
if `moveNext` fails to move the token sequence (end of stream), the `tokenId` will remain the same (see below) and the wile will loop endlessly. Maybe not because of `offset` check, but still.

Reply:
In last commit had handled any tokensequence overshoot isue